crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1232,6 +1233,50 @@ fn handle_trailing_expression_starred_star_end_of_line_comment<'a>(
     CommentPlacement::leading(starred.as_any_node_ref(), comment)
 }
 
+/// Handles trailing own line comments before the `as` keyword of a with item and
```

Comment:
i wonder if an abstraction with two nodes with a single token in between makes sense, which would also force us to handle them consistently. We have the same pattern in slices (`x[a:b]`), argument lists `x(a,b)` and binary expressions (`a+b`). (also this feature is inconsistent because `with (a as b)` works but `except (a as b)` doesn't).

Reply:
And dicts and dict comprehensions and walruses

Reply:
https://github.com/astral-sh/ruff/issues/5780